src/Bicep.Decompiler/TemplateConverter.cs

```diff
@@ -613,10 +613,16 @@ private SyntaxBase ParseString(string? value, IJsonLineInfo lineInfo)
             }
         }
 
-        private static IntegerLiteralSyntax ParseIntegerJToken(JValue value)
+        private static ExpressionSyntax ParseIntegerJToken(JValue value)
         {
-            var ulongValue = value.Value<ulong>();
-            return SyntaxFactory.CreateIntegerLiteral(ulongValue);
+            var longValue = value.Value<long>();
+            if (longValue < 0)
```

Comment:
I think you want to use the same logic as here: https://github.com/Azure/bicep/blob/2d31817c59e57fd50be8b17025894f9a7b583869/src/Bicep.LangServer/Handlers/InsertResourceHandler.cs#L286-L293. I think `(ulong)longValue` here should in fact be `(ulong)-longValue`.

I think it would also be worth creating a method in `SyntaxFactory` that can be used in both places rather than duplicating the logic.

Reply:
You are right about that! The CreateNegativeIntegerLiteral() function wants the positive value and then it will add the MinusToken to make it a negative value in essence.

Good idea, I will create the method!